### ISSUE TITLE
AWACS Alpha (0.0.18) Discussion

### ISSUE THREAD

**Issue by Applevangelist** [FlightControl-Master/MOOSE#1715]
Discussion AWACS Alpha 0.0.2

**Comment by thebgpikester** [comment]
AI Considerations:
Player dispatching AKA "the mission"
- How do we determine the threat axis and where to tell players to hold CAP?
- How can we determine what range from what point to launch an Intercept or release from CAP?

Scenario1: AI CAP on station, enemy planes approach (something?) within X miles
Scenario2: Player CAP on station or assigned, enemy planes approach (something?) within X miles
Scenario3: We don't use AI CAPs and we don't have a player signed in. When do we launch an AI intercept?
Scenario4: We Dont use AI CAP but a player is checked in on the ground to simulate scramble. When do we give him a scramble/intercept?

Things to consider....
Anchor zones might equal the threat defence range
Multiple anchor zones or cap zones
Not having CAP in the config 
Having GCI Quick reaction
How to assign a mission to an AI or a Player.
Multple players.
Player created Anchor zones via Markpoint?
Threat axis (makes for nicer hold points)
Different ranges for different reactions Bogey inside X range of X asset like awacs or air base or high value asset.

**Comment by Applevangelist** [comment]
> 

Let's park the GCI / Scramble features for the moment, maybe later, but may show up as request
Human tasking will be preferred to AI

**Comment by thebgpikester** [comment]
WORKFLOW SUGGESTIONS
Expanding **checkin** as both a functional "Asset" for the AI and an immersion item:
Checkin is normally done via a single frequency where flights will authenticate to a controller who will assign them to a a senior Weapons Director in charge of that area. The mission and picture and lowdown are provided, check in frequency is provided and the flight is pushed to a specific controller frequency. This manages multiple traffic into channels that are manageble.

Typically the flow is:
Authentication is probably out of scope. The pilots own voice is assumed, so on pressing the 'check in' button the sample conversation flow could be:

"Uzi 1-1, Magic, Radar contact, bullseye xxx/xx at x thousand. Say when ready for Picture."
Menu item: "Ready" (can we wipe the other options?)
"Picture is (Clean/Single group/2 groups/multiple groups), Track X BULLSEYE 123, 45, Angels 6, HOSTILE, Track X BULLSEYE 123 45 Angels 6, HOSTILE" etc. I dont think it makes sense to try to do any complex PICTURES but I will have a think about if that's even possible.
End that with a custom "LOWDOWN" if the mission design enters one for this mission which is a text sentence like "Situation is to execute current no fly zone, say when ready to copy station coordinates"
Menu item: "Ready" (can we wipe the other options?)
"Uzi 1-1, Magic, New Mission. CAP/Intercept/Scramble. Northings 12345, Eastings 056789, 20 thousand, Hot leg zero four five. Time on station 30"

"Check in with DarkStar on 123 decimal 45"

If planes don't check in with AWACS then all messages should be advisories. For the sake of making a workable script I dont think it's neccessary to put any additional conversation on this channel, but its possible.

At this point player should change channel to the assigned directors channel as a new AI asset. Pictures can now be done away with because they are cumbersome and we can assume tactical control which is a direct realtionship with specific flights. Meaning we can do BRAA's. Services can now be shown on menu like Tripwire, bogey dope, check out. POPUP calls can alert specific players to a new detection inside 50nm. 

**Assignments/Missions**. These as you say can be parked for later iterations but would be seen as AI tasking player.

Targetting: I would see the following directive from AWACS:
"Uzi 1 1, DarkStar , Target group track..n BRAA 123 45 angels 6, hot, hostile" 
Create two new menu items:
JUDY
UNABLE
Put the target assignment on 1 minute repeats. 
"uzi 11, DarkStar , roger unable/Judy".
The AWACS would cease if the pilot uses menu item Judy or unable. The difference is that the AWACS would assign a GCI or alternative flight if unable. This will silence the callouts once the player has the target or repeat until the plane is dead.

Once dead some workflow to get the play back on mission:
"Uzi 11, DarkStar , confirmed grand slam, vectors to station 123 45"

Check out:
"Uzi 11, DarkStar , roger, check in with Magic on 123 decimal 45, good day/good night, safe flight, thanks etc
-removes flight from being tasked

AWACS 
"Magic,All stations, SUNRISE SUNRISE SUNRISE, Magic"
--AWACS takes off and begins service

"Magic, All stations, SUNSET, SUNSET, SUNSET, MAgic"
--AWACS stops service (RTB?)

**Comment by thebgpikester** [comment]
Group naming 
Normally you would name groups something easy like NORTH GROUP, HEAVY GROUP, DISTINCTIVE NAME GROUP. The issue with the sim is that people will flood the airspace and the naming still needs to be useful to the ear, but you run out of compass directions quickly.

I think with the INTEL that you could provide group labels as unique in any way that makes sense, for example, you can use the alphabet 

ALPHA
BRAVO
CHARLIE

These would stick in the head better and give maximum 25 rotating names to use for groups, but it might need some additions to INTEL, perhaps to create friendly names for groups or clusters

**Comment by thebgpikester** [comment]
**Using lua Code and MOOSE to describe a realistic PICTURE.**
For this I studied [](https://armypubs.army.mil/epubs/DR_pubs/DR_a/ARN33396-ATP_3-52.4-000-WEB-1.pdf)
A Picture is used to convey the relevant parts of the entire airspace the contorller (and pilots) are assigned to.

The basics of a PICTURE is the CORE INFORMATION. Core information is a simple list of bogeys (and up) bullseyes that groups of planes are currently at. CORE INFORMATION is used when the picture is just not making sense from a tactical or organisation perspective, it just reads out the BULLSEYEs of the top three biggest threat bogeys. It's used when you can't anchor the picture on what looks like a singles groups tactical threat - i.e. just crap flying around. 

ANCHORING is when there is a definite groups of interest - the most significant threat is called out and the second two groups are given with relationships to that one, in Range or azimuth. For example, there's ten groups flying around. Two of them are flying towards the awacs together, side to side. This would then make sense to anchor the picture on the most threatening of those two contacts and the callout would be "two groups, azimuth (number of miles apart), cardinal group, bullseye etc..."

After considering the code level there are some simple decisions becoming apparent.
- The times when ANCHORING the picture in DCS are restricted to very infrequent occasions where in multiplayer humans can move in two groups more than 3nm apart together, or two flights performing structured tactics like brackets and more complex formations of many flights like Champagne or stingers. Its basically highly uncommon. The second type of groups moving "in concert" would be AI and I cannot think of an occasion where this has every been coded. It may happen by accident only if the lead and wingman of a group become split by over 3nm only.
- It's so much easier to create a paralel of an AWACS defined GROUP and a DCS defined GROUP of units. It's a massive simplification and vulnerable to confusion but it fits well to do call outs on a DCS GROUP scale. However, in design this needs to be considered if it is viable. If two planes are split by 3nm, which is seen with AI, then using the DCS group could be detrimental and confusing. I would challenge this should be unit based using clusters and 3nm on the group leader.
The most common probably picture would be "Magic, Picture, two groups CLOSING, RANGE 10, South group bullseye 1-2-3 forty-five, 10 thousand, track south, hostile, north group 1-2-3 fifty-five, ten thousand, track south, hostile, both groups fulcrum." 
TBD? You would still have to work out if there was a group that was worth anchoring on.

Therefore if we only include CORE INFORMATION in our PICTURES, the PICTURE is very simplified and the following formats and technical work is required:

Basics of a large picture:
DARKSTAR, TEN GROUPS. GROUP BULLSEYE 020/25, TWENTY-SEVEN THOUSAND, TRACK EAST, HOSTILE, HEAVY, THREE CONTACTS. GROUP BULLSEYE 270/25, FIFTEEN THOUSAND, TRACK WEST, HOSTILE. GROUP BULLSEYE 290/35, TEN THOUSAND, TRACK WEST, HOSTILE.” 

Friendlies, strangers, civilians, non participating flights are not counted in the picture.
Groups are aircraft sharing the same 3nm bubble.

The format is "C/S. <number of groups>, GROUP BULLSEYE (zero two five), Twenty-five, Seventeen thousand, HOSTILE, tracking, ID, Fillers. <repeat up to three>


**TRACK**
Use the track to cardinal function. One exception is you can use CAP or CAPPING, ANCHORED or MANOUEVER, but this is hard to code so here is a suggestion.
Capture the heading twice before messaging. If the heading has changed then we are GROUP MANOUVER else we are TRACK NORTH/s/e/w

**DECLARATION coding**
DCS AWACS automatically terms all coalition opposites as HOSTILE. This is done wiht the AWACS TASK and for datalink, the ELPRS task on the AWACS DCS group. It is regrettibly stupid to copy anything else as the datalinks will be marked as hostile by AWACS, rather than ambiguous HAFU. Thus, all contacts are de facto HOSTILE and bandit, friendly, spades, or complex ID cannot be used in game. However, I would not hardcode this into the speech lines because of several reasons, most importantly we could add SPADES, BOGEY and so on in two future cases:
1) When the IFF scripting is incorporated
2) If or when we implement a cold war 


**FILLER CODING**
Fillers are additional information and their inclusion or exclusion requires code and effort. Here are some rules:
1) Avoid too many if it expands the message length too much (how much is too much)
2) Order of preference is:
- number of contacts - single is assumed so omit, two-ship, HEAVY 3+
- type using NATO name like FLANKER, FULCRUM, FISHBED, FLOGGER, FOXBAT
- HIGH (over 40K)
- FAST/VERY FAST (m1.0+)/(m1.5+)
- UNTARGETED (depends on if we are tracking assignments, but we should)
- TARGETED BY/LEANING ON/THREAT TO
This is very difficult to code, but if a heading is in collision with another friendly group you can use 'leaning on' and callsign to specifiy who is being chased.

**Overall code design**
There must be a polygon area for the AO for these pictures to make sense in. We have to cut this up so we can make some distinctions of "relevant" and "no factor" This enables us to cut noise from awacs, tankers and such.
We need a threat axis. AWACS is a good start but it needs a direction, probably from awacs to an anchor point.
We have code for what is and isnt a fighter. We can use this to only zone in on fighters. We can also use this to develop a higher threat, we have that. Remember if we have bombers and fighters mixed we have to list out fighters first and nearest to furthest along the threat axis.
picture should be called from menu on demand, and provided on new groups that are relevant like fighters. It's given on check in too.
We have to count the awacs groups (not the DCS groups) using INTEL class. These are in the AO which probably shoudlnt be stretched deep to avoid counting support planes.
Fighters inside a threshold should then be anchored if less than 3 groups, else read out as CORE INFORMATION with nearest hot one first.
We can hardcode the TYPE for the filler at a certain range to simulate NCTR or offboard secret stuff.
We have to do a lot of range code. Take the groups that INTEL gives, get nearest, check the third nearest etc. Its goign to be nasty to come up with algorithms that handle the anchoring or go for unrelated core information, but it it can be done its soemthing no one else can do.

**Comment by Applevangelist** [comment]
Thanks @thebgpikester I'll go through that tomorrow. I have implemented a basic picture and limited to INTEL clusters, with a disctinction in or out of the AO. For the base picture your described most elements are there in code.

**Comment by thebgpikester** [comment]
The most important thing is that you only need the three most relevant contacts for a core information picture as per the book. The simulation does it wrong, it just keeps saying ADDITIONAL GROUP BULLS...and repeat.
The proword additional is not used unless someone is pointing out you missed something. So basically much more simple until you have to pick "which three". I would say the three nearest to the awacs but thats a decision to consider, perhaps three nearest fighters that are also in the AO instead.

**Comment by Applevangelist** [comment]
Yep. Have a delimitation for that, no worries

**Comment by dMARLAN** [comment]
Example of a full intercept just for reference (btw this is really exciting to have this developed for DCS)

Example Callsigns
AIC = BEAR
FTR = COLT

**Pre-Commit**
PRI: "BEAR, TWO GROUPS, 
     GROUP, BULLSEYE 300/50, 25 THOUSAND, CAP, HOSTILE.
     GROUP, BULLSEYE 340/70, 30 THOUSAND, CAP, HOSTILE."

- repeated every 60 seconds unless there is an important change in aspect.

**Commit**
PRI: "BEAR, TWO GROUPS, 
     GROUP, BULLSEYE 300/50, 25 THOUSAND, TRACK SOUTH, HOSTILE.
      RECOMMEND COMMIT."

- AIC will anchor only the highest threat group.
- AIC can recommend or direct commit.
- Fighters can call their own commit even without recommendation/direction.

PRI: "COLT, COMMIT."

- AIC will allow a small pause (5-10 seconds) for fighters to perform administrative tasks such as examples below
AUX: "COLT, CHECK TAPES, MASTER ARM."
AUX: "COLT, DEPLOY."  or "COLT, PUMP" as briefed/appropriate.
AUX: "COLT, TAC LEFT, 350, GO."

**Post-Commit**
PRI: "BEAR, 2 GROUPS, RANGE, 20. 
     LEAD GROUP, BULLSEYE 300/50, 25 THOUSAND, HOT, HOSTILE.
     TRAIL GROUP, BULLSEYE 340/70, 30 THOUSAND, HOT, HOSTILE."

**Targeting**
- Fighters will target themselves into the groups, targeting must be assigned no later than the pre-briefed MELD range otherwise the AIC will directively target the fighters.

PRI: "COLT, TARGET LEAD GROUP. BEAR, MONITOR TRAIL GROUP."
PRI: "BEAR."

**Tac Range**
- At Tac Range, and informative call will be made indicating that MELD is less than a minute away, this call is typically made by AIC, but in the case that AIC does not transmit tac range call, any fighter with SA will make the call (typically the lead).

PRI (AIC): "COLT, LEAD GROUP, 55 MILES."

**Meld**
- Meld is the time when radar sanitization is changed to gaining & maintaining radar about the targeted group. If the group was already detected during sanitization then it is only an informative part of the intercept, otherwise "clean" calls and talk ons can be performed.

AUX: "COLT, MELD."

**Sort**
- Fighters will sort targets inside their respective targeted group as per their pre-briefed sorting trees. Typically this starts with azimuth sort and requires no comms. If the fighter lead determines an azimuth sort is not possible he will switch to first range then elevation.

AUX: "COLT, SORT RANGE."
AUX: "COLT 12, SORTED TRAIL."

**Employment**
- Fighters will employ their missiles on timeline.

PRI: "COLT 11, FOX 3."
AUX: "COLT 11, CRANK LEFT."

PRI: "COLT 12, FOX 3."
AUX: "COLT 12, CRANK RIGHT."

**Defense**
- Fighters will defend on timeline and make time of missile last support calls.

PRI: "COLT 11, SKATE SOUTH, PITBULL, LEAD MAN, LEAD GROUP."
PRI: "COLT 12, SKATE SOUTH, PITBULL, TRAIL MAN, LEAD GROUP."

------
Can get into a lot more depth but that's the rough idea of an intercept timeline.

As for how this applies to a scripted AIC here are the important parts:

Pre-Commit, Broadcast Control, Bullseye Format
- AIC transmits every 60 seconds the most relevant groups using cardinal track directions, and never including more than the 3 most important groups. Based on the pre-briefed ranges, the picture shall be considered "CLEAN" if threats are outside that briefed range. For example, any groups beyond 120 NM from a friendly asset will not be included in the broadcast.

Commit, Broadcast Control, Bullseye Format
- AIC will direct or recommend a commit by retransmitting a broadcast call, but only anchoring 1 group (still say total number of groups)
- Fighters can also simply call commit at any time before being directed or recommended.

Post-Commit, Tactical Control, Bullseye Format
- AIC will give fighters a brief time to complete administrative actions, then the AIC will label the picture in tactical control, meaning group labels are applied, and aspect is now relative to the fighters.

Tac Range
- AIC will make the Tac Range call at the pre-briefed range.

Meld
- At Meld range is when AIC will transition to BRAA Format.
- AIC will typically not talk after Meld except for some exceptions, such as threat groups, untargeted groups, etc.
- Inside Meld range is where the AIC can now directively target fighters, typically when fighters are dragging or if an emergent situation arises.

e.g. (PRI) "COLT 11, BEAR, TARGET, POP-UP GROUP, BRAA 050/30, 10 THOUSAND, HOT, HOSTILE"
      (PRI)"COLT 11."

Other
- AIC will call when groups are passing/crossing
- AIC will call group maneuvers
e.g. "BEAR, LEAD GROUP, BEAM EAST."
e.g. "BEAR, LEAD GROUP, MANEUVER."  "BEAR, LEAD GROUP, DRAG NORTH."
- Fighters can request BRAA/BOGEY DOPE at any time.
e.g. "BEAR, COLT 11, BRAA, TRAIL GROUP."
     "COLT 11, BEAR, TRAIL GROUP, BRAA 010/40, 25 THOUSAND, HOT, HOSTILE."



Gotta run now, but hopefully this helps. It's a very generalized post, if you guys have any questions please let me know, I'd love to help get this up and running. I've been learning this stuff for a long time and have the luxury of knowing both an E2 NFO and a retired F-18 pilot who has helped me understand a lot of this.

The document thebgpikester linked is fantastic, but also give a look at the P-877 ADV E2 NFOTS, it's also Public Release and a wonderful resource for understanding AIC.

**Comment by dMARLAN** [comment]
Got some more time, for maybe a first edition instead of trying to code an algorithm that will be able to label groups could just start with perhaps "first group" "second group" etc. (for tactical control only, in broadcast control it would always just be "group") then perhaps a future edition could attempt to automatically label groups appropriately.

Comm Menu commands that could be used:

Commit (Changes the AI AIC from Broadcast control to Tactical control)
BRAA/Bogey Dope
Picture
Separation (By default could be between the first and second groups, or could have a sub menu with a list of all groups in the picture)
     e.g. "BEAR, COLT 11, SEPARATION SECOND GROUP."
     "COLT 11, BEAR, SECOND GROUP, SEPARATION 15, 25 THOUSAND, HOT, HOSTILE."
Judy (Not really used IRL, but for an AI that could be annoying in some cases, handy to have)

Settings (either in code, or menus)
Broadcast Range - Maximum range between a fighter and a bogey/bandit that the AI AIC will consider. Anything beyond this range will be excluded.

Tac Range - Range where Tac Range call will be made.

Meld Range - Range where meld occurs. AIC transitions to BRAA format here. Groups detected inside meld range are considered pop-up groups.

Threat Range - Range where groups are considered threat groups.

**Comment by Applevangelist** [comment]
> Deleted this. Only need to worry about labelling if we program an actual PICTURE with anchoring. Stick to "GROUP" and leave it. There may be additional descriptors of POPUP GROUP, THREAT GROUP, but that's it. Labels are done when anchoring the picture only. Anchoring I will talk about in general.

I have added labels when a new contact comes in from intel. Wrt the below posts question is when to call them out

**Comment by Applevangelist** [comment]
That’s a lot of information, thanks guys. We need to simplify that somewhat and bring into diagrams to make it work. Note: AI CAP callouts. ATM I only plan to have them on voice for checking in, checking out, arriving at station. Possibly can also add also get a text out when going on an Auftrag, when this Auftrag is successful, and when they RTB (bingo, tos over, winchester, dead/crash/eject) or go back to station. Everything else AIC to general audience and to Player comms.

**Comment by dMARLAN** [comment]
What kind of diagrams do you need?

Aside from trying to label group names it can actually be fairly simple.

1) AIC in Broadcast Control, Bullseye Format.
-> Broadcast all relevant groups, no more than 3, and nothing outside of broadcast range.

2) Fighter activates "Commit" menu option
-> AIC changes to Tactical Control, Bullseye Format. Labels groups with "first group" "second group" etc. based on range. This isn't a realistic practice but perfectly usable for a first iteration and less awful than the DCS infinite additional groups.

3) AIC makes Tac Range call.

And that's about it for the generic timeline, AIC can make pop-up/threat group callouts if a group does suddenly appear inside meld/threat range during a commit and there are those menu options to request things from the AIC but they're all pretty straight forward I think.

**Comment by Applevangelist** [comment]
> Example of a full intercept just for reference (btw this is really exciting to have this developed for DCS)
> 
> Example Callsigns AIC = BEAR FTR = COLT
> 
> **Pre-Commit** PRI: "BEAR, TWO GROUPS, GROUP, BULLSEYE 300/50, 25 THOUSAND, CAP, HOSTILE. GROUP, BULLSEYE 340/70, 30 THOUSAND, CAP, HOSTILE."
> 
> * repeated every 60 seconds unless there is an important change in aspect.
> 
> **Commit** PRI: "BEAR, TWO GROUPS, GROUP, BULLSEYE 300/50, 25 THOUSAND, TRACK SOUTH, HOSTILE. RECOMMEND COMMIT."
> 
> * AIC will anchor only the highest threat group.
> * AIC can recommend or direct commit.
> * Fighters can call their own commit even without recommendation/direction.
> 
> PRI: "COLT, COMMIT."
> 
> * AIC will allow a small pause (5-10 seconds) for fighters to perform administrative tasks such as examples below
>   AUX: "COLT, CHECK TAPES, MASTER ARM."
>   AUX: "COLT, DEPLOY."  or "COLT, PUMP" as briefed/appropriate.
>   AUX: "COLT, TAC LEFT, 350, GO."
> 
> **Post-Commit** PRI: "BEAR, 2 GROUPS, RANGE, 20. LEAD GROUP, BULLSEYE 300/50, 25 THOUSAND, HOT, HOSTILE. TRAIL GROUP, BULLSEYE 340/70, 30 THOUSAND, HOT, HOSTILE."
> 
> **Targeting**
> 
> * Fighters will target themselves into the groups, targeting must be assigned no later than the pre-briefed MELD range otherwise the AIC will directively target the fighters.
> 
> PRI: "COLT, TARGET LEAD GROUP. BEAR, MONITOR TRAIL GROUP." PRI: "BEAR."
> 
> **Tac Range**
> 
> * At Tac Range, and informative call will be made indicating that MELD is less than a minute away, this call is typically made by AIC, but in the case that AIC does not transmit tac range call, any fighter with SA will make the call (typically the lead).
> 
> PRI (AIC): "COLT, LEAD GROUP, 55 MILES."
> 
> **Meld**
> 
> * Meld is the time when radar sanitization is changed to gaining & maintaining radar about the targeted group. If the group was already detected during sanitization then it is only an informative part of the intercept, otherwise "clean" calls and talk ons can be performed.
> 
> AUX: "COLT, MELD."
> 
> **Sort**
> 
> * Fighters will sort targets inside their respective targeted group as per their pre-briefed sorting trees. Typically this starts with azimuth sort and requires no comms. If the fighter lead determines an azimuth sort is not possible he will switch to first range then elevation.
> 
> AUX: "COLT, SORT RANGE." AUX: "COLT 12, SORTED TRAIL."
> 
> **Employment**
> 
> * Fighters will employ their missiles on timeline.
> 
> PRI: "COLT 11, FOX 3." AUX: "COLT 11, CRANK LEFT."
> 
> PRI: "COLT 12, FOX 3." AUX: "COLT 12, CRANK RIGHT."
> 
> **Defense**
> 
> * Fighters will defend on timeline and make time of missile last support calls.
> 
> PRI: "COLT 11, SKATE SOUTH, PITBULL, LEAD MAN, LEAD GROUP." PRI: "COLT 12, SKATE SOUTH, PITBULL, TRAIL MAN, LEAD GROUP."
> 
> Can get into a lot more depth but that's the rough idea of an intercept timeline.
> 
> As for how this applies to a scripted AIC here are the important parts:
> 
> Pre-Commit, Broadcast Control, Bullseye Format
> 
> * AIC transmits every 60 seconds the most relevant groups using cardinal track directions, and never including more than the 3 most important groups. Based on the pre-briefed ranges, the picture shall be considered "CLEAN" if threats are outside that briefed range. For example, any groups beyond 120 NM from a friendly asset will not be included in the broadcast.
> 
> Commit, Broadcast Control, Bullseye Format
> 
> * AIC will direct or recommend a commit by retransmitting a broadcast call, but only anchoring 1 group (still say total number of groups)
> * Fighters can also simply call commit at any time before being directed or recommended.
> 
> Post-Commit, Tactical Control, Bullseye Format
> 
> * AIC will give fighters a brief time to complete administrative actions, then the AIC will label the picture in tactical control, meaning group labels are applied, and aspect is now relative to the fighters.
> 
> Tac Range
> 
> * AIC will make the Tac Range call at the pre-briefed range.
> 
> Meld
> 
> * At Meld range is when AIC will transition to BRAA Format.
> * AIC will typically not talk after Meld except for some exceptions, such as threat groups, untargeted groups, etc.
> * Inside Meld range is where the AIC can now directively target fighters, typically when fighters are dragging or if an emergent situation arises.
> 
> e.g. (PRI) "COLT 11, BEAR, TARGET, POP-UP GROUP, BRAA 050/30, 10 THOUSAND, HOT, HOSTILE" (PRI)"COLT 11."
> 
> Other
> 
> * AIC will call when groups are passing/crossing
> * AIC will call group maneuvers
>   e.g. "BEAR, LEAD GROUP, BEAM EAST."
>   e.g. "BEAR, LEAD GROUP, MANEUVER."  "BEAR, LEAD GROUP, DRAG NORTH."
> * Fighters can request BRAA/BOGEY DOPE at any time.
>   e.g. "BEAR, COLT 11, BRAA, TRAIL GROUP."
>   "COLT 11, BEAR, TRAIL GROUP, BRAA 010/40, 25 THOUSAND, HOT, HOSTILE."
> * AIC will call out arms, which is when a group (contacts inside 3nm of each other) separate.
>   "BEAR, LEAD GROUP, MANEUVER AZIMUTH. NORTH ARM 25 THOUSAND FLANK NORTH WEST. SOUTH ARM 25 THOUSAND FLANK SOUTH WEST."
> 
> Gotta run now, but hopefully this helps. It's a very generalized post, if you guys have any questions please let me know, I'd love to help get this up and running. I've been learning this stuff for a long time and have the luxury of knowing both an E2 NFO and a retired F-18 pilot who has helped me understand a lot of this.
> 
> The document thebgpikester linked is fantastic, but also give a look at the P-877 ADV E2 NFOTS, it's also Public Release and a wonderful resource for understanding AIC.

Ranges - these are zones so I can get filtering right. What are typical sizes, thinking of concentric rings as easiest example I have atm


OpsZone - 120 NM - all detected contacts are stored by AIC. Considered for picture, sorted by distance to AO center or bulls, max callout 3


AO or TacZone - Polygon or round - contacts in this zone are considered for bogey dope and tasking. Sorted by distance to CAP plane max callout 3

Merged - TBD - when going through the contacts anyway, callout merges

Popup group - basically a new contact callout where the contact is first seen in the AO

Meld range - would only apply to groups where AIC knows they are committed

Declare - AIC will help IFF (dependent on era setting) to identify closest contact to a cap in a 3nm range around the plane. Note - need to come up with a menu so the player can vid and feed back the intel to the AIC.

**Comment by Applevangelist** [comment]
> What kind of diagrams do you need?
> 
> Aside from trying to label group names it can actually be fairly simple.
> 
> 1. AIC in Broadcast Control, Bullseye Format.
>    -> Broadcast all relevant groups, no more than 3, and nothing outside of broadcast range.
> 2. Fighter activates "Commit" menu option
>    -> AIC changes to Tactical Control, Bullseye Format. Labels groups with "first group" "second group" etc. based on range. This isn't a realistic practice but perfectly usable for a first iteration and less awful than the DCS infinite additional groups.
> 3. AIC makes Tac Range call.
> 
> And that's about it for the generic timeline, AIC can make pop-up/threat group callouts if a group does suddenly appear inside meld/threat range during a commit and there are those menu options to request things from the AIC but they're all pretty straight forward I think.

From a Software Design perspective the FSM does cyclic rounds of functions, so I need to think about a flow of things to be done one after another based on certain states. Player menu requests can bei single events (like a bogey dope call) which however doesn’t necessarily change the data state - but a commit to a task would change a data item and in the next cycle lead to a different behaviour. 

So in the base status round we have a couple of things to consider:
- is the AIC alive, on station, and/or is there a shift change needed (basically done, apart from when the thing’s dead)
- did we request enough AI CAP and in what state are they? (WIP, they go on their stations)
- Next up would be - regular picture calls. I have an on demand one, need to make this a loop every  5/10? Minutes
- Then merge calls, tripwire, but need to check every round -> towards players
- Tasking will be the main challenge; AI can be send on an Auftrag and will do their thing; Auftrag FSM sends back event based, so I don’t need to check every time
- Players - here the tasking is a matter of how tightly we want to check on what they are actually doing, I can imagine ppl calling commit and then shooting the wrong targets etc.

**Comment by Applevangelist** [comment]
Before I forget - has anyone a lua table ready for dcs unit type names to Nato types (Foxbat, Flanker…) conversion?

**Comment by Applevangelist** [comment]
https://en.m.wikipedia.org/wiki/List_of_NATO_reporting_names_for_fighter_aircraft

**Comment by thebgpikester** [comment]
I believe inside a specific zone, we should have a basic state of **targeted** or **untargeted**. The dispatching and control of the intercepts is a primary AI job. 

**TRIGGERS** Out of band events
New Enemy plane in the AO area (AO defined by polygon and must be not all encompassing so as to make it responsive.
Existing Enemy plane inside a friendly planes tactical range (e.g. 45nm) This can be an autocommit logic:
- if player and player targeted then issue THREAT call logic
- if player and player untargeted then issue COMMIT call logic
- Targeted contact has left AO or VANISHED 
    - issue player with a "New mission, return to station" or "INVESTIGATE BULLSEYE 123 45"
    - issue AI with RTB
 - Untargeted contact has left AO or vanished
 - Targetee (the plane issues with the job of intercepting) is dead > Switch contact to untargeted so the loops restarts.   
 - certain dead events on any team, I think callouts from AWACS in BULLSEYE format for group bullseye 123 45 vanished would be uyseful for everyone.

**Dispatching flow** iterate contacts and check for:

Overview: match contacts with a targeting directive to player and AI and dispatch and monitor AUFTRAG for non player.

if the contact in a zone which I have to do something about it at all? 
 is there an existing AUFTRAG in progress on the contact?
 if the contact already targeted from a prior cycle by an alive group? > issue a BRAA call and break out the loop
  if there is a human resource (prefered) that is checked in? > Tasking function
  else No> launch AUFTRAG (intercept)

**Player pre commit flow**
Pre check in we do nothing menu shows options to check in
Post check in we add the player to a resource pool and replace check in with check out.
AWACS checks his CAP zones, sends mission on station using the existing code



**Player Post commit flow**
(- triggers monitor lives above)
- call BRA on 30s intervals - now this is where it might get interesting. There is a condition where another group within your threat range comes into your bubble during the interecept and the question is how to deal with that, especially if there are multiple threat groups. If you create a new intercept based on the distance you sent the first intercept, you will get an escalation and rolling hot war that goes quite mental. TBD
- JUDY optional from menu to silence this.
- untargetted filler becomes higher priority

- i find it hard to create a flow beyond single engagments that makes sense at this point.

Thats all for now.  ssince its very complex and I can't think of everything properly yet.

**Comment by Applevangelist** [comment]
> I think inside a specific zone, we should have a basic state of **targeted** or **untargeted** for all red coalition planes. The dispatching and control of the intercepts is a primary AI job that is intertwined with Players. This should work with and without players by default out of the box I think.
**Indeed a planned feature** 
> 
> **TRIGGERS** Out of band events that oiccur requiring some action:
> 
>     * New Enemy plane in the AO area (AO defined by polygon - must not be huge).
** DONE for contact saving, listing. Question on how/if to prevent mission designers to create stupid zones ... **
> 
>     * Existing Enemy plane inside a friendly planes tactical range (e.g. 45nm) This can be an autocommit logic:
>
>     * if player and player targeted then issue THREAT call logic
> 
>     * if player and player untargeted then issue COMMIT call logic
** Can be done in the Tripwire loop, messaging depending on what the player & mission/target status is **

>     * Targeted contact has left AO or VANISHED
>       
>       * issue player with a "New mission, return to station" or "INVESTIGATE BULLSEYE 123 45"
>       * issue AI with RTB
>     * Untargeted contact has left AO or vanished
> 
>     * Targetee (the plane issues with the job of intercepting) is dead > Switch contact to untargeted so the loops restarts.
> 
>     * certain dead events on any team, I think callouts from AWACS in BULLSEYE format for group bullseye 123 45 vanished would be uyseful for everyone.
** right **
> 
> **Dispatching flow** iterate contacts and check for:
> 
> Overview: match contacts with a targeting directive to player and AI and dispatch and monitor AUFTRAG for non player.
> 
> if the contact in a zone which I have to do something about it at all? is there an existing AUFTRAG in progress on the contact? if the contact already targeted from a prior cycle by an alive group? > issue a BRAA call and break out the loop if there is a human resource (prefered) that is checked in? > Tasking function else No> launch AUFTRAG (intercept)
> 
> **Player pre commit flow** Pre check in we do nothing menu shows options to check in Post check in we add the player to a resource pool and replace check in with check out. AWACS checks his CAP zones, sends mission on station using the existing code
> 
> **Player Post commit flow** (- triggers monitor lives above)

** Here it gets tricky, and I think it makes sense to think about having a control SRS Radio Freq per checked in player group **

>     * call BRA on 30s intervals - now this is where it might get interesting. There is a condition where another group within your threat range comes into your bubble during the interecept and the question is how to deal with that, especially if there are multiple threat groups. If you create a new intercept based on the distance you sent the first intercept, you will get an escalation and rolling hot war that goes quite mental. TBD

** Good one. Having though about this, could be a couple of cases, basic one a Gorilla or Bomber group just crossing the zone. Maybe check type and flight altitude planes. AI Fighter - would they meddle in the fight? Not sure. Enemy players - can't really make an automated decision. For the latter 2 types can make sense to auto-add them to the TARGET object tracked with the specific mission, avoiding overhead with JUDY etc, but would mean an auto-commit. Other option - generally ignore - also for further taskign by other AI/Player - until left the intercept bubble.

>     * JUDY optional from menu to silence this.
> 
>     * untargetted filler becomes higher priority
> 
>     * i find it hard to create a flow beyond single engagments that makes sense at this point.
** keeping it simple - the idea is to have a flow of single engagements.
> 
> 
> Thats all for now. ssince its very complex and I can't think of everything properly yet.

**Comment by thebgpikester** [comment]
Today and yesterday currently playing with various PICTURE calls. Enjoying the complexity of how to provide the top three groups of a bigger picture, and how to weight the ways we assess with human eyes the relevant parts. I am using the following criteria:
- Range - the obvious and cheap ordering system
- Aspect - I think its possibly the most important part of assessing threat directionally.
- Fighter or other. I think over three group pictures can easily filter non fighters
- Group size. Heavy groups are always dangerous

There can be more but when figuring out a table sorting algorithm you really can't get an effective ordering unless there is somehow a weighting system and the order is interchangeable. When speaking to Frank he set me up with a basic sorting algorithm and it is never going to be as good as human eye:

Given
```
 contact={name=grp:GetName(), threat=grp:GetThreatLevel(), dist=dist}
 
 local function sort(a,b)
  return **(a.dist<b.dist)-- or (a.dist==b.dist and a.threat>b.threat)**
end
```

Like SQL it will sort one way then resolve differences which is really pants.
Would like to hear your ideas here. Objective to give the top three contacts of importance in the picture

**Comment by Applevangelist** [comment]
Range - the obvious and cheap ordering system
    Aspect - I think its possibly the most important part of assessing threat directionally.
    Fighter or other. I think over three group pictures can easily filter non fighters
    Group size. Heavy groups are always dangerous

Looks like a sensible approach to me. 
A bit of a bucket approach might help.
* First, filter if in AO to reduce size. 
* Second - Aspect hot on the AO can be an additional factor here, maybe with a threshold of distance?
* Third - group size. If we deem heavy groups dangeorus, we need to up their threatlevel, as e.g. a bomber pack comes back twith TL 8 oder so, and a single fighter with TL10. Getting the overall TL might be an option as groups can be max 4(?) in size. Fi

**Comment by thebgpikester** [comment]
Right now I'm using the AWACS as 'centre of the universe' to consider threat and range, but this should probably be script configurable. For example use a unit or static or airbase as a centrally pinned vulnerable point. Note that this threat distance is not going to be refered to ever outside of code, its arbitrary evaluation (and subject to argument).
Carrier, AWACS, airfield, city, topographical point, EWR installation can all be examples of a defensive point the algorithm can decide on.

So developing on your feedback fill 3 range buckets
<50nm,<100nm, >100nm

Then I think a combined threat level works like:
group size * grp:GetThreatLevel()

And finally
ordered by aspect?

I'm gonna aim for this and see how it calls out.

**Comment by Applevangelist** [comment]
> AWACS as 'centre of the universe' to conside
* The coded version is currently either bulls eye or the center of the AO, as it is the target to keep the AO clean. But yeah, for threat assessment can be any coordinate. Too many options might create problems down the road with noob mission designers, maybe we stick to one approach first
* On the aspect calc I need to read up the source, I think we currently need a directional vector for that, which is only provided with positionables.

**Comment by Applevangelist** [comment]
Heading can be provided to the coordinate, ok, so we need a direction set in which the defense coord is “looking” (threat axis). Let me add that and make it visual on the map

**Comment by thebgpikester** [comment]
Aspect was the reason for the requirmeent of a centre point to reference from. But you can hard code it to the awacs, because as said, its not the aspect that is read out when doing a BRAA and BULLSEYE has no aspect
local aspect = tgtCoord:ToStringAspect(currentCoord)
This will only be for differentiating high threat fighters when painting a picture, i don't recomemnd its usage for anything else. The BRAAA code is pretty easy and has that.

**Comment by Applevangelist** [comment]
The downside is that the awacs is changing headings when orbiting so the aspect changes. I can have that in a shadow coord so it doesn’t influence bra callouts

**Comment by thebgpikester** [comment]
The aspect of a contact remains the same whichever way the awacs is facing. It's not Angle on Bow :). If the ToStringAspect() is incorrectly writte (i thought it was OK previously) then this needs some different code. A bandit driving towards the awacs will always be hot.

**Comment by Applevangelist** [comment]
Need to cross check this

**Comment by Applevangelist** [comment]
function aspectcheck()
  local myposition = ZONE:FindByName("NW Zone"):GetCoordinate()
  local group = GROUP:FindByName("Aerial-1")
  local groupcoord = group:GetCoordinate()
  local groupheading = group:GetHeading()
  groupcoord.Heading = groupheading
  local aspect = groupcoord:ToStringAspect(myposition)
  MESSAGE:New(string.format("Aerial-1 is aspect %s",aspect),10,"Info"):ToAll()
end

local timer = TIMER:New(aspectcheck)
timer:Start(10,15)

**Comment by Applevangelist** [comment]
Works if I add the heading

**Comment by Applevangelist** [comment]
Fixed low down in group and positionable

**Comment by Applevangelist** [comment]
> Right now I'm using the AWACS as 'centre of the universe' to consider threat and range, but this should probably be script configurable. For example use a unit or static or airbase as a centrally pinned vulnerable point. Note that this threat distance is not going to be refered to ever outside of code, its arbitrary evaluation (and subject to argument). Carrier, AWACS, airfield, city, topographical point, EWR installation can all be examples of a defensive point the algorithm can decide on.
> 
> So developing on your feedback fill 3 range buckets <50nm,<100nm, >100nm
> 
> Then I think a combined threat level works like: group size * grp:GetThreatLevel()
> 
> And finally ordered by aspect?
> 
> I'm gonna aim for this and see how it calls out.

Implemented now in the code. Need to change getting contacts from contact to cluster in a meaningful way - contacts are easier to address for many things, but the cluster mechanism from intel can club stuff together, eg a mainstay and it’s escort. I’ll have a think on that

**Comment by dMARLAN** [comment]
Additional/Pop-up/Threat range would be based on the fighter positions BTW for future versions.

It doesn't really make sense to have it hard set to an airbase/AWACS/etc.

For example, You may consider the AA-10C Missile a category SAR-2 threat with a 25 NM threat range and lesser missiles a SAR-1 threat with a 15 NM threat range.

Therefore if a MiG-29 was within 25 NM of a friendly fighter that would be a THREAT call

"COLT 11, BEAR, THREAT GROUP, BRAA 250, 23, 25 THOUSAND, HOT, HOSTILE."

But if it was, say, a MiG-23, it wouldn't be until 15 NM.

Both of these ranges are in regards to the fighter itself, it wouldn't have much use if a bandit was 25 NM from the AWACS, but 100 NM from the fighter (well, except being a THREAT to the AWACS of course)

**Comment by Applevangelist** [comment]
Since I'm working on missile callouts today, IMHO an option would be to to go by seeker type, else I'd need an extensive list of missiles and their ranges ... considering: https://wiki.hoggitworld.com/view/DCS_Class_Weapon
Callouts will be for

```
Weapon.MissileCategory
  AAM          1
  SAM          2
```
 
and then maybe distance per type

```
Weapon.GuidanceType
  IR                   2 -- 5nm
  RADAR_ACTIVE         3  --25nm
  RADAR_SEMI_ACTIVE    4  --15nm 
  RADAR_PASSIVE        5  --10nm
```

With SAM types getting a 25nm range. For the THREAT call, a fixed distance might be ok for a first approach (~25nm)

**Comment by Applevangelist** [comment]
I have an era switch - non-moderns don't get a callout ;) - just a dcs thing, good to know, but helpful for Players. Same I'll have some "Vector to target BRA" callouts for stationing and intercepts for players, I'll make both switchable

**Comment by thebgpikester** [comment]
Only two comments from the initial start. I ran the demo and i was doing everything I had hoped.

"All stations, Magic, clear" 
should be
"All stations, Magic. Picture clean" Clean is specific to radar returns and without the proword 'picture', it feels a bit wonky when said (often).

Bogey Dope with no nearest target
"Jedi- four, Magic, clear"
should be 
"Jedi-four, Magic, clean" (or unable)

BRAAs and callout format
Some of the callouts are using older code because you get a mix of old and new wording types for BRA and some fo the newer ones are wrong too.

three examples:
![image](https://user-images.githubusercontent.com/22999891/166115577-3e6d8c20-9683-4152-be33-b82cc7ac64a4.png)


Most of the miles in new examples are being read out as seperate digits. So, "1 2 3 for 064" should be "1 2 3, sixty-four miles"

First example should be 
"Magic, Dodge 3 1, commit echo group, brah 3 2 1, sixty-three miles, 25 thousand"

Second example should be
"Foxtrot group, Rock 0 3 2, 64 miles, Twenty-five thousand, Cold, Targeted by Dodge 1 1"
"Echo group, Rock 3 5 5, 78 miles, Twenty-five thousand, 2-ship, <type>, Targetted by Dodge 3 1"

Third example is the old Flight control BR, should be;
"Dodge 1 1, Magic, vector to station 2 1 4, thirty-five miles, Angles 22

I got one script error and it was a GetCordinate failure
2022-04-30 17:16:11.260 INFO    SCRIPTING:  18500( 21645)/E:                          UNIT02235.GetCoordinate({[1]=Cannot GetCoordinate,[Positionable]={[UnitName]=Aerial-1#001-01,[ClassName]=UNIT,},})
2022-04-30 17:16:11.260 INFO    SCRIPTING: Error in SCHEDULER function:[string "C:\Users\Pikes\AppData\Local\Temp\DCS.openbeta\/~mis00007103.lua"]:21647: attempt to index local 'FirstUnitCoordinate' (a nil value)

I know this one as it needs an error check if exists before checking the group:GetCoordinate() because they die and mess stuff up.

This is awesome work, i cant believe its so close!

**Comment by Applevangelist** [comment]
Great feedback, and easy to fix :)

**Comment by thebgpikester** [comment]
On to some more :D sorry

AI checkin. Seems to be per unit instead of per flight. There's a lot of checking in on the same callsign.

**Comment by thebgpikester** [comment]
AWACS:New() -6th parameter is super weird.
In the docs this is an area. I changed it to "Georgia" and it began in messages that AWACs was Georgia control and all seemed good.

Then it appeared to be used as a bullseye and with a single point which is counterintuitive. The instructions say

Name of the round, mission editor created Fighter Engagement operations zone (FEZ) this AWACS controls. Can be passed as #ZONE_POLYGON. Will be used in referenc call, ensure a radio friendly name that does not collide with NATOPS keywords.
Did this cause error   **SCRIPTING:  14356(  8795)/E:                    COORDINATE01238.MarkupToAllFreeForm(ERROR: Currently a free form polygon can only have 10 points in total!)** 
I instinctively tried to mimmic a border and I dont know what I should do with it.

Can we check what this reference call is about and what the intention is of this area?

**Comment by thebgpikester** [comment]
testawacs:SetBorderZone(ZONE:FindByName("Blue Border"))

Related to the above, its another border, not sure its purpose and what it dictates. Documentation is empty. Is this INTEL related?

Ultimatley when designing a mission I want to draw a border for ignoring contacts and a border for the zone the AWACS controls, both of which will be multiple points.
If there needs to be a single coordiante, I'd rather it was seperate and explained than try too hard to multipurpose all these poly zones into single poitns also?

**Comment by thebgpikester** [comment]
Actually, scratch that, the 6th parameter seems to be a custom bullseye AND a zone of operations and the callouts are given to a zone. I put the bulleseye out to sea and its bascally disregarded. I think this is an area requiring a bit of rework. I'm OK with custom bullseyes and ops zones, but we have to have a centre of a zone made precisely. A Bullseye is already positioned and provides a coordinate, so i dont see that we need to change it. Instead soemthing like
AWACS:CustomBullseyeName = "string"

This way the bulseye position is used, retained and if you need a zone for something else in parameter 6 then use it that way.

**Comment by thebgpikester** [comment]
![image](https://user-images.githubusercontent.com/22999891/166120247-59615a56-c847-4b1e-9f90-121e7c647b45.png)
Cadence sounds better in TTS with
All stations, Magic. Picture. 3 groups.
Theres a comman rather than the period after stations.

**Comment by thebgpikester** [comment]
![image](https://user-images.githubusercontent.com/22999891/166120301-93414ef5-c374-4fde-8fb4-b326fb7a3404.png)
These are great callouts. I think atmospherically a callout more similar to the DCS one is nicer and you can do some randomisation and expand this with the following suggestions:
All stations, Magic ... <-- this opener again in broadcast
chute spotted/<C/S> vanished/friendly flight faded/lost contact with C/S/CSAR Scramble
Bullseye 1 2 3 forty-five miles

**Comment by dMARLAN** [comment]
I am probably missing something but I am not understanding the purpose of "All Stations", AWACS calls made in Broadcast Control are already addressed to everyone "Magic, 3 groups, group, bullseye ......" and doesn't need "picture" unless it was a response to a request from a fighter because in broadcast control every broadcast is a (new) picture.

e.g. Broadcast Control (Picture):
[Magic]: "Magic, 3 groups, group, bullseye 250/50, 25 thousand, cap, hostile.
group, bullseye 300/80, 30 thousand, cap, bogey spades.
group, bullseye, 200/30, 15 thousand, track west, bogey spades."

e.g. Post-Merge Picture call from Fighter: (Of course we don't need the flow/state but just an example of real comm)
[Colt 11]: "Magic, Colt 11, flow east, picture."
[Magic]: "Magic, Picture clean, flow miami, say state."

Regarding a downed friendly fighter, the comm brevity is RIGHTGUARD with location. This indicates a fighter (or any friendly flight) is downed and SAR should be initiated if able.

e.g. 
[Magic]: "Magic, RIGHTGUARD, RIGHTGUARD, Bullseye X/Y"

**Comment by thebgpikester** [comment]
> I am probably missing something but I am not understanding the purpose of "All Stations", AWACS calls made in Broadcast Control are already addressed to everyone "Magic, 3 groups, group, bullseye ......" and doesn't need "picture" unless it was a response to a request from a fighter because in broadcast control every broadcast is a (new) picture.
> 
> e.g. Broadcast Control (Picture): [Magic]: "Magic, 3 groups, group, bullseye 250/50, 25 thousand, cap, hostile. group, bullseye 300/80, 30 thousand, cap, bogey spades. group, bullseye, 200/30, 15 thousand, track west, bogey spades."
> 
> e.g. Post-Merge Picture call from Fighter: (Of course we don't need the flow/state but just an example of real comm) [Colt 11]: "Magic, Colt 11, flow east, picture." [Magic]: "Magic, Picture clean, flow miami, say state." [Colt 11]: "Colt 11, tiger."
> 
> Regarding a downed friendly fighter, the comm brevity is RIGHTGUARD with location. This indicates a fighter (or any friendly flight) is downed and SAR should be initiated if able.
> 
> e.g. [Magic]: "Magic, RIGHTGUARD, RIGHTGUARD, Dodge 41, Bullseye X/Y"

Its got legacy origins, its probably better to use the way you said. Not a hard and fast design, its changed over the years of course, this would come froma time before these books :)

**Comment by Applevangelist** [comment]
> On to some more :D sorry
> 
> AI checkin. Seems to be per unit instead of per flight. There's a lot of checking in on the same callsign.

Ah that’s per group ATM, each a single flight. So Dodge 1, 2, 3. Maybe I do a little round robin eith group names to make it more distinctive

**Comment by Applevangelist** [comment]
> AWACS:New() -6th parameter is super weird. In the docs this is an area. I changed it to "Georgia" and it began in messages that AWACs was Georgia control and all seemed good.
> 
> Then it appeared to be used as a bullseye and with a single point which is counterintuitive. The instructions say
> 
> Name of the round, mission editor created Fighter Engagement operations zone (FEZ) this AWACS controls. Can be passed as #ZONE_POLYGON. Will be used in referenc call, ensure a radio friendly name that does not collide with NATOPS keywords. Did this cause error **SCRIPTING: 14356( 8795)/E: COORDINATE01238.MarkupToAllFreeForm(ERROR: Currently a free form polygon can only have 10 points in total!)** I instinctively tried to mimmic a border and I dont know what I should do with it.
> 
> Can we check what this reference call is about and what the intention is of this area?

I you look into the ME, Rock#POLYGON_ZONE is the name of the Helo making the FEZ zone ;) Moose picks that up and creates a Polyzone on loading; it’s a little used Sven feature :)

**Comment by Applevangelist** [comment]
> Actually, scratch that, the 6th parameter seems to be a custom bullseye AND a zone of operations and the callouts are given to a zone. I put the bulleseye out to sea and its bascally disregarded. I think this is an area requiring a bit of rework. I'm OK with custom bullseyes and ops zones, but we have to have a centre of a zone made precisely. A Bullseye is already positioned and provides a coordinate, so i dont see that we need to change it. Instead soemthing like AWACS:CustomBullseyeName = "string"
> 
> This way the bulseye position is used, retained and if you need a zone for something else in parameter 6 then use it that way.

Yep that’s arguably a more natural approach for the mission designer.

**Comment by Applevangelist** [comment]
> ![image](https://user-images.githubusercontent.com/22999891/166120247-59615a56-c847-4b1e-9f90-121e7c647b45.png)
> Cadence sounds better in TTS with
> All stations, Magic. Picture. 3 groups.
> Theres a comma rather than the period after 'stations' that makes the sentence more natural.

Actually mostly I generate 2 texts, one for the screen and one for TTS. In TTS the dots work better for natural speech, readability on screen output is not that good however. And yeah, need to redo the old BRA calls to systematically fit. Must say however that TTS outputs 321° nicely as 3 2 1

**Comment by Applevangelist** [comment]
Most of the commentary should be resolved next time around

**Comment by DesignsOAK** [comment]
Does AWACS:_StartIntel(awacs) effectively delay call outs until the AWACS arrives at its assigned orbit zone?  I still would like to delay the AWACS calling sunrise from the ground.

Also, the SRS implementation is a bit different than MOOSE's STTS.TextToSpeech - really need a control parameter for the STTS volume (the default full volume is way too loud).  Thanks for adding the Google TTS credentials, but parameter #5 does not seems to be accepting the Google TTS voice (example: "en-GB-Wavenet-F").  Has anyone else got Goggle TTS to work?

One more - does AI-AWACS inherit INTEL?  Is it possible to filter out certain target types, like helos below a given altitude for example?

**Comment by Applevangelist** [comment]
The sunrise call is delayed until we reach 10k feet. I‘m using Moose MSRS, need to check if there’s a volume option, too. The parameters are handed to the SRS tts component exe in the SRS basefolder, so whatever parameter is there should be available/ can be made available. Helos are filtered by default ATM indeed

**Comment by Applevangelist** [comment]
> Also, the SRS implementation is a bit different than MOOSE's STTS.TextToSpeech - really need a control parameter for the STTS volume (the default full volume is way too loud). Thanks for adding the Google TTS credentials, but parameter #5 does not seems to be accepting the Google TTS voice (example: "en-GB-Wavenet-F"). Has anyone else got Goggle TTS to work?

-- Reading up on Google, the voice will be set in the payload json if I'm not mistaken

**Comment by thebgpikester** [comment]
Minor point on Bullseye again, its correct for the miles most times but one of the functions reads out the miles with a leading zero which makes the number read out as zero-six-nine (069). I've seen both versions used so much be multiple methods going on. Drives my ears wild when hearing "zero-four-five, zerosix-nine miles" instead of the more often "zero-four-five, sixety-nine miles"
Its not used in the standard BRA or bulleye used, I heard it early on. I think its the picture call out;
![image](https://user-images.githubusercontent.com/22999891/166995407-90abc6e8-b1c4-45fc-9a32-3ad1dafe2500.png)


Magic calls out pictures before sunrise :)

Magic used words Bee-Arrr-Ayyy instead of "BRAH"

Magic still uses the group size before bullseye on nearly all callouts. eg. 
![image](https://user-images.githubusercontent.com/22999891/166997258-74eff1bf-74ac-44e6-80d6-31bd75d28a9c.png)

**Comment by thebgpikester** [comment]
FEATURE request:
GCI control without awacs as a physical plane.

Not everyone has one, the UK has awacs envy, I think we are "in between purchases" :P I dont think anything needs to change from the process side.

**Comment by thebgpikester** [comment]
Feature request
text on/off toggle

Yes i know its excellent for debugging, but I wanna actually have a go!

**Comment by thebgpikester** [comment]
FEATURE request documentation

I think the FEZ/OPSZone/Bullseye thing is too hard to comprehend its usage. Its a combined thing, custom bullseye, it doesnt use the bullseye from the DCS Mission, It has a large circular zone for which not sure what that does and a smaller polygon zone, in red, which seems "serious" but is also not understandable. There is also the defensive zone.
1) Whats the circle of the opszone? Detection? call outs? Pictures? engagements?
2) Whats the FEZ? For engagements? Outside is ignored? Anything else special happen if an enemy is in it?
3) Why is the bullseye coupled to this? Why do you have to have the bullseye in the middle of the FEZ? What can't have the bullseye elsewhere and uncoupled from this zone?
4) Defensive zone. Seems like its a trigger for something to happen, but its not explained what exactly? If something enters that, what happens?
5) These polygon zones can they have more than 10 points?

I think these answers would go a long way to help someone know how to setup the class

**Comment by thebgpikester** [comment]
DEFECT: Magic will answer a check in when he is dead :)
Although this might be useful for having GCI only missions there shoudl be some detection that AWACS is alive. Perhaps a function that allows it to speak?


function AWACSisOnDuty()
--is it alive, airborne, over 10000ft high return true else return false
end

if AWACSisOnDuty() then TTS("im talking because im at work")

**Comment by DesignsOAK** [comment]
> The sunrise call is delayed until we reach 10k feet. 

Correct, but the AWACS should not be making any picture/threat calls before "Sunrise" and it is still doing so while on the ground.  Pike's AWACSisOnDuty() check may do the trick.  Just a thought.

**Comment by DesignsOAK** [comment]
DEFECT: on first 2 picture calls the AWACS points out 2 groups:
![image](https://user-images.githubusercontent.com/70976717/167057094-00f49220-05ac-4031-9d17-a60783d91ac9.png)

However, the voice version of this says "Olympus one one two zero, ninety eight miles" and then "Olympus one one three six, eighty miles".  Note the 4 digit bearings.

**Comment by Applevangelist** [comment]
As I predicted everyone totally gets hung up on the callouts ;)

**Comment by Applevangelist** [comment]
> "zero-four-five, zerosix-nine miles"
fixed
> Magic used words Bee-Arrr-Ayyy instead of "BRAH"
> depends on google vs win I think. The Win version says "brah" on "BRA", Google spells that out... interesting

**Comment by Applevangelist** [comment]
> > The sunrise call is delayed until we reach 10k feet.
> 
> Correct, but the AWACS should not be making any picture/threat calls before "Sunrise" and it is still doing so while on the ground. Pike's AWACSisOnDuty() check may do the trick. Just a thought. - Rowdy

Need to cross check that. I do do a height check before starting the voice queue

**Comment by Applevangelist** [comment]
> DEFECT: on first 2 picture calls the AWACS points out 2 groups: ![image](https://user-images.githubusercontent.com/70976717/167057094-00f49220-05ac-4031-9d17-a60783d91ac9.png)
> 
> However, the voice version of this says "Olympus one one two zero, ninety eight miles" and then "Olympus one two three six, eighty miles". Note the 4 digit bearings. - Rowdy

fixed next round

**Comment by Applevangelist** [comment]
> Feature request text on/off toggle
> 
> Yes i know its excellent for debugging, but I wanna actually have a go!

Done next time, you get a function for that

**Comment by Applevangelist** [comment]
> FEATURE request: GCI control without awacs as a physical plane.
> 
> Not everyone has one, the UK has awacs envy, I think we are "in between purchases" :P I dont think anything needs to change from the process side.

Maybe

**Comment by Applevangelist** [comment]
> FEATURE request documentation
> 
> I think the FEZ/OPSZone/Bullseye thing is too hard to comprehend its usage. Its a combined thing, custom bullseye, it doesnt use the bullseye from the DCS Mission, It has a large circular zone for which not sure what that does and a smaller polygon zone, in red, which seems "serious" but is also not understandable. There is also the defensive zone.
> 
>     1. Whats the circle of the opszone? Detection? call outs? Pictures? engagements?
> 
>     2. Whats the FEZ? For engagements? Outside is ignored? Anything else special happen if an enemy is in it?
> 
>     3. Why is the bullseye coupled to this? Why do you have to have the bullseye in the middle of the FEZ? What can't have the bullseye elsewhere and uncoupled from this zone?
> 
>     4. Defensive zone. Seems like its a trigger for something to happen, but its not explained what exactly? If something enters that, what happens?
> 
>     5. These polygon zones can they have more than 10 points?
> 
> 
> I think these answers would go a long way to help someone know how to setup the class

It does use the BE from the mission editor. Initially you only need to worry about a) FEZ b) move the blue/red BE around c) where the AWACS should be orbiting and d) where to put the 1st station for the CAP flights. Maybe e) name your reference.

**Comment by Applevangelist** [comment]
> DEFECT: on first 2 picture calls the AWACS points out 2 groups: ![image](https://user-images.githubusercontent.com/70976717/167057094-00f49220-05ac-4031-9d17-a60783d91ac9.png)
> 
> However, the voice version of this says "Olympus one one two zero, ninety eight miles" and then "Olympus one two three six, eighty miles". Note the 4 digit bearings. - Rowdy

fixed

**Comment by thebgpikester** [comment]
You can drop singleton as its 'implied' like hot, to save space. That's cool, it was an old function but yes.

The cadence of the callouts is fast and being computerised its not quite spot on with all the pauses, so the clues like "miles" are more like helpers although its a reasonable ask to get these closer and omit, I think you would have to try it out, my concern is we turn this into unintelligible garbage if we miss adding helper clues, but am willing to try :) The thing is, they condensed this entire thing over the years, its much shorter now than twenty years ago and shorter even than 40 years where they would have used miles, so there is a definite "dating" that can be attributed.

Labels make no sense unless the call is anchored, in DCS no sense can be made of multiple groups working together in range or azimuth because multiple groups of AI dont work together. So the Picture has to have some give and take in order to be workable hence the weird labelling of a rotating alphabet to give them a character and an arbitrary listing of the nearest three groups to awacs. It's a sacrifice, as soon as we can generate AI tactics in formations of champagnes and boxes and such we can make callouts of 2 groups range 10 south group blah blah...

The order is still wrong though, the group size is leading the bullseye call in the pictue, it should be a filler at the end. ANd the older function with th eleading zero is used for miles in pictures, that's wrong, other than that there's compromise.

You can use this as a reference for the speed and content but this is a labelled two group east-west fight 
https://drive.google.com/file/d/0BwUQri1lYjQ8eUk3MncxTG1pWXc/view?usp=sharing&resourcekey=0-P6h_tFQnjAWMex6JKjbVZQ 

There's defintie space to remove and simplify, just have to maintain the intelligibility with the AI voice.

**Comment by thebgpikester** [comment]
> As I predicted everyone totally gets hung up on the callouts ;)

You have absolutely no idea what is about to be unleashed and you will thank your lucky stars that you got a chance to tighten it up because when you get hit by 2000 armchair Tom Cruises wanting to fix something, its gonna be painful. This is the sharp end of nerd city!

**Comment by Applevangelist** [comment]
Lol. Absolutely. I feel I have already 2 nerds on me … sizing of groups has been moved around. Need to drop the singleton still

**Comment by thebgpikester** [comment]
still miles called as digits in one of the picture fuctions.
Google not working out of the box, still searching for the solution.
I think patrol "replacement" and GCI could be thought about now this is quite far down the line.
Still think non AWACS and Non fighter squadron version is important.
Looking good!

**Comment by thebgpikester** [comment]
_still miles called as digits in one of the picture fuctions. -- yes, if using Desktop TTS._
Now I tried Google I see the issue. Google voice does the right miles and it actually sounds better But she does start the degrees: 045 as Oh-four-five rather than zero four five. So i get there are issues here. The cadence is MUCH better in the waveform picked. It's fast and clear. 

Feedback on the voices.

Pretty awesome immersion factor when you have the different voices talking. It's a definite "wow factor". I was thinking about cheap ways to build on this. Perhaps you can get the callsigns from the group and use that as a hash value to slightly alter the voices? I saw you did something like this with the nationalities, i.e. the French plane had French voice. Maybe each callsign can get enumerated at startup and the speed and nationality and sex can be randiomised? It might end up even more awesome if that were possible

**Comment by Applevangelist** [comment]
Currently it’s a list of 10 voices which are randomly taken; French guy in the mirage is luck ;)

**Comment by Applevangelist** [comment]
Oh and the uk voice says oh five two, a us one will say zero five two

**Comment by thebgpikester** [comment]
Bon chance!
So, small one again on zones.
![image](https://user-images.githubusercontent.com/22999891/168309783-ab1e6c7b-409c-415d-a3e6-69c5b1686108.png)
Reference this image, if you overlap the big green circle from the centre of the FEZ with the hpoly that extends out, the pair of Flankers do not get picked up until inside the circular group, which wasnt my expectations, I thought It would be 
1. Inside the AWACS unit detection, but outside the rejection zone first.
2. Inside the FEZ regardless
3. Inside the defense zone.

I'm still unsure how these zones are supposed to work, they dont really make sense to my brain at least.

**Comment by Applevangelist** [comment]
When they are inside the radar zone or inside the red ops zone, they should be picked as targets. I'll cross-check that as a test. Forget about the radar zone and the zone around the AIC, these are "internal use"

**Comment by thebgpikester** [comment]
The second they crossed that green circular line, she offered "New group..." The Flanker gorup out westr really dont often get picked up. But the Blindfires tend to get followed for a long time. Coincidence?

**Comment by Applevangelist** [comment]
Not really. I haven't made the FEZ bigger than my radar circle, so I left out a check on the FEZ for the immediate announcement of new contacts, also need to check INTEL on the acceptzone setup I made. Good hint!
